### PR TITLE
#160452470 Integrate Coveralls code coverage service

### PR DESCRIPTION
**Pull request by Alameen688**
#### What does this PR do?
Integrate Coveralls code coverage service
#### Description of Task to be completed?
- Link coveralls to the github repo
- Add coveralls badge to readme file
- Configure travis to run coverage after test has passed
#### How should this be manually tested?
- Clone repo or pull latest changes
- Checkout to `ch-160452470-coveralls-setup`
```bash
> git checkout ch-160452470-coveralls-setup
```
- Install packages 
```bash 
 > npm install
```
- Run coverage script 
```bash
> npm run coverage
```
- Coverage report should be logged on the command line
- [Coveralls build output](https://coveralls.io/builds/19222433)
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#160452470](https://www.pivotaltracker.com/story/show/160452470)
#### Screenshots (if appropriate)
- N/A
#### Questions:
- None

### CODE REVIEW

**Comment by hnobi** [comment]
Nice job, it run the test coverage but then this errors showed up in my CLI 

`[error] "2018-09-25T18:36:05.275Z"  'error from lcovParse: ' 'Failed to parse string'
[error] "2018-09-25T18:36:05.276Z"  'input: ' ''
[error] "2018-09-25T18:36:05.277Z"  'error from convertLcovToCoveralls'
`

**Comment by Alameen688** [comment]
> Nice job, it run the test coverage but then this errors showed up in my CLI
> 
> `[error] "2018-09-25T18:36:05.275Z" 'error from lcovParse: ' 'Failed to parse string' [error] "2018-09-25T18:36:05.276Z" 'input: ' '' [error] "2018-09-25T18:36:05.277Z" 'error from convertLcovToCoveralls'`

It's because there is no test for any part of the code. Therefore, what is being passed into coveralls is an empty coverage report.
I've added a test to show that it works. I'll push soon.

**Review by uchemukolo — COMMENTED** [review]
looks good to me

**Comment by hnobi** [comment]
> > Nice job, it run the test coverage but then this errors showed up in my CLI
> > `[error] "2018-09-25T18:36:05.275Z" 'error from lcovParse: ' 'Failed to parse string' [error] "2018-09-25T18:36:05.276Z" 'input: ' '' [error] "2018-09-25T18:36:05.277Z" 'error from convertLcovToCoveralls'`
> 
> It's because there is no test for any part of the code. Therefore, what is being passed into coveralls is an empty coverage report.
> I've added a test to show that it works. I'll push soon.

Noted

File: .babelrc [hunk 1]
```diff
@@ -4,7 +4,7 @@
             "targets": {
                 "node": "current"
             },
-            "plugins": ["transform-object-rest-spread"]
+            "plugins": ["transform-object-rest-spread", "istanbul"]
```
Comment (omedale) [inline .babelrc]: I thought we are using `nyc` , why do we have `istanbul` in your PR @Alameen688

Comment (Alameen688) [reply .babelrc]: It's a plugin needed for babel support. However, we do not have any use case for it now. I have removed it as requested.

**Pull request merged**